internal/boxcli/run.go

```diff
@@ -122,8 +125,12 @@ func runScriptCmd(cmd *cobra.Command, args []string, flags runCmdFlags) error {
 	envOpts := devopt.EnvOptions{
 		OmitNixEnv: flags.omitNixEnv,
 		Pure:       flags.pure,
+		RecomputeEnv: &devopt.RecomputeEnvOpts{
+			Disabled:              !flags.recomputeEnv,
+			StateOutOfDateMessage: fmt.Sprintf(devbox.StateOutOfDateMessage, "with --recompute=true"),
```

Comment:
A few questions/comments:

* Is this message needed at all? It feels like if someone is voluntarily using --recompute=false, they know what they want. 
* This is copy pasted a bunch of times, maybe create a `var` for this?
* If this message is not needed, maybe inside devbox package it can simply skip the warning if `StateOutOfDateMessage` is empty. That way you only have to set this field in one place (in shellenv)